src/FixerDefinition/VersionSpecification.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\FixerDefinition;
+
+/**
+ * @author Andreas Möller <am@localheinz.com>
+ *
+ * @internal
+ */
+final class VersionSpecification implements VersionSpecificationInterface
+{
+    /**
+     * @var int|null
+     */
+    private $minimum;
+
+    /**
+     * @var int|null
+     */
+    private $maximum;
```

Comment:
>the problem here would be if we would convert sth to binary on our fixer.

Not only in that case.
If we have a fixer that would deal with `(binary)` we want to add an example here, but if `(binary)` is no longer valid on a high PHP version the example won't lint and so fails.
Same for the PHP4 constructors, if those would be removed (i.e. no longer lint) on PHP8, the fixer can only give valid samples for versions below that.